codec.go

```diff
@@ -113,3 +119,33 @@ func (c *Codec) HardwareConfigs(dt HardwareDeviceType) (configs []CodecHardwareC
 	}
 	return
 }
+
+func (c *Codec) HasHardwareConfigMethodFlag(chcmf CodecHardwareConfigMethodFlag) bool {
```

Comment:
I'd rather not have this function in the lib, it seems this logic could be achieved with `(c *Codec).HardwareConfigs()` with relatively little lines of codes and is present in the `hardware decoding` example

Reply:
Oh i see there is an issue with "HardwareConfigs" because the devicetype is never used. We should remove the Argument and Update the hw example